### DISCUSSION TITLE
Warning sign with red and/or orange feeds?

### DISCUSSION THREAD

**Discussion by gingerbeardman** [FreshRSS/FreshRSS#4359] [Q&A] [answered]
I just started using FreshRSS, importing an OPML from Reeder. My plan is to use a variety of RSS apps rather than being bound to one and FreshRSS as my "main" source of feeds.

Anyway, I am seeing the following and I do not know how serious this is:

- ⚠️ warning signs
- 🟥 red feeds
- 🟧 orange feeds

Questions:
- What do these mean? 
- Where can I view the error details?
- How is it recommended to resolve these?

Many thanks for any assistance!

### Background 

The relevant feeds are feeds from websites that have been operating for decades and that did not pose a problem with Reeder or other desktop/mobile RSS apps.

![Screen shot 2022-05-10 at 15 07 58](https://user-images.githubusercontent.com/49612/167648487-7c2fbd5c-42f2-45d9-b750-7fd8c5447a03.png) ![Screen shot 2022-05-10 at 15 20 25](https://user-images.githubusercontent.com/49612/167651214-39ba0f7f-3fcb-4213-b9db-da4ec23621f5.png)

**Comment by gingerbeardman** [comment]
I have managed to repair the following:

- IA: superpaint
- IA: ultrapaint

These two had truncated URLs in the OPML (problem in the OPML generator, not FreshRSS)

Also:

- Accidentally in Code
- Daring Fireball
- furbo.org
- inessential
- kottke.org
- One Foot Tsunami
- NewNewsWire
- The Shape of Everything

I did a feed validation on the base URL to get the latest feed URL, changed it on the manager screen, and finally refreshed the feeds.

Example fix:

- https://inessential.com/feed.json

to

- https://inessential.com/xml/rss.xml

Q: why did I have to update these 8 feeds?

**Comment by Alkarex** [comment, marked as answer]
Hello,
Yellow/orange means empty feed, red means error when retrieving the feed.
In case of error, first thing to do is to try to validate the feed via https://validator.w3.org/feed/

**Reply by gingerbeardman** [reply]
Thanks, switching to slightly different feeds did the trick, there were some validation warnings on those before I looked up the latest feeds.

Does not validate:
https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Finessential.com%2Ffeed.json

Validates:
https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Finessential.com%2Fxml%2Frss.xml

So it's interesting to me that FreshRSS is more strict that other apps with regards to feed validity. All good!

**Reply by Frenzie** [reply]
Presumably that JSON feed is valid enough, but we don't support JSON feeds at this time.

**Reply by gingerbeardman** [reply]
> Presumably that JSON feed is valid enough, but we don't support JSON feeds at this time.

Ah, that's it! I believe the feeds I had to replace were all JSON.

Mystery solved.

Thanks!
